### PR TITLE
Remove apache commons logging in favor of slf4j + log4j for JAVA 9

### PR DESCRIPTION
**Pull request by Siedlerchr**
While testing with Java 9, I noticed that commons logging infrastructure we use is not compatible.
However,  [sfl4j](https://www.slf4j.org/) + [log4j2](https://logging.apache.org/log4j/2.x/) are already java 9 compatible.

Added ADR

----

- [ ] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [ ] Screenshots added (for bigger UI changes)
- [x] Manually tested changed features in running JabRef
- [ ] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -138,7 +138,8 @@ dependencies {
     compile 'org.jsoup:jsoup:1.11.2'
     compile 'com.mashape.unirest:unirest-java:1.4.9'
 
-    compile 'commons-logging:commons-logging:1.2'
+    compile 'org.slf4j:slf4j-api:1.8.0-beta0'
```
Comment (koppor) [inline build.gradle]: Please comment that 1.8.0 is required for Java 9 support.

File: docs/adr/0002-use-slf4j-for-logging.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# Use slf4j together with log4j2 for logging
+
+* Currently JabRef uses apache commons logging 1.2 for logging errors and messages. However, this is not compatible with java 9 and is superseeded by log4j.
```
Comment (koppor) [inline docs/adr/0002-use-slf4j-for-logging.md]: "Up to version 4.1"

File: docs/adr/0002-use-slf4j-for-logging.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+* Log4j is already defined as dependency and slf4j has already been required by a third party dependency
+
+## Considered Alternatives
+
+* [Log4j2](https://logging.apache.org/log4j/2.x/)
+* [SLF4J](https://www.slf4j.org/)
+
+
+## Decision Outcome
+
```
Comment (koppor) [inline docs/adr/0002-use-slf4j-for-logging.md]: Please place the decision here. :-)

The next subsection is for a detailed elaboration.

File: docs/adr/0002-use-slf4j-for-logging.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+## Pros and Cons of the Alternatives <!-- optional -->
+
+### *Log4j2*
+
+* `+` *Already defined*
+* `+` *Java 9 support*
+* `-` *Direct dependency*
+* *[...]* <!-- numbers of pros and cons can vary -->
```
Comment (koppor) [inline docs/adr/0002-use-slf4j-for-logging.md]: Remove this line

File: docs/adr/0002-use-slf4j-for-logging.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+## Considered Alternatives
+
+* [Log4j2](https://logging.apache.org/log4j/2.x/)
+* [SLF4J](https://www.slf4j.org/)
+
+
+## Decision Outcome
+
+## Pros and Cons of the Alternatives <!-- optional -->
```
Comment (koppor) [inline docs/adr/0002-use-slf4j-for-logging.md]: Remove `<!-- optional -->`

File: docs/adr/0002-use-slf4j-for-logging.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+### *Log4j2*
+
+* `+` *Already defined*
```
Comment (koppor) [inline docs/adr/0002-use-slf4j-for-logging.md]: Please remove the stars. They are only meant as placeholders...

File: docs/adr/0002-use-slf4j-for-logging.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
+# Use slf4j together with log4j2 for logging
+
+* Currently JabRef uses apache commons logging 1.2 for logging errors and messages. However, this is not compatible with java 9 and is superseeded by log4j.
+* SLF4J provides a facade for several logging frameworks, including log4j and supports already java 9
+* Log4j is already defined as dependency and slf4j has already been required by a third party dependency
+
+## Considered Alternatives
+
+* [Log4j2](https://logging.apache.org/log4j/2.x/)
+* [SLF4J](https://www.slf4j.org/)
```
Comment (koppor) [inline docs/adr/0002-use-slf4j-for-logging.md]: I think, the options are:

- Log4J2
- SLF4J with Logback binding
- SLF4J with Log4J binding

Because of https://logging.apache.org/log4j/2.0/log4j-slf4j-impl/, it seemed, you chose the Log4J binding

**Comment by Siedlerchr** [comment]
The breaking build is still from the test I fixed in the other pr

**Comment by koppor** [comment]
Thank you for using [MADR](https://adr.github.io/madr/) 🎉.

**Comment by koppor** [comment]
Switching to logback could maybe solve https://github.com/JabRef/jabref/pull/3511?

File: docs/adr/0002-use-slf4j-for-logging.md [hunk 8]
```diff
@@ -1,31 +1,43 @@
...
+* We chose slf4j with log4j2 binding, because it only requires minimal changes to our logging infrastructure and it is claimed that  
+> Apache Log4j 2 is an upgrade to Log4j that provides significant improvements over its predecessor, Log4j 1.x, and      provides many of the improvements available in Logback while fixing some inherent problems in Logback’s architecture.
+* Furthermore, as slf4j is a facade for several loggers, the underlying implementation can easily be changed in the future
+
+
+## Pros and Cons of the Alternatives 
+
+### Log4j2
+
+* `+` Already defined
```
Comment (koppor) [inline docs/adr/0002-use-slf4j-for-logging.md]: Do you mean "already defined as dependency"? --> Maybe rephrase to "Dependency already exists"

**Comment by lenhard** [comment]
Codacy still complains about an unused import, but this is a false positive, right? Otherwise, checkstyle would complain. If this is not a problem, then this PR can be merged.

**Comment by Siedlerchr** [comment]
Interesting case. the import was only used because it was referenced in an javadoc comment. So I removed it.
